src/inmanta/deploy/state.py

```diff
@@ -36,15 +38,45 @@ def requires_view(self) -> Mapping[ResourceIdStr, Set[ResourceIdStr]]:
     def provides_view(self) -> Mapping[ResourceIdStr, Set[ResourceIdStr]]:
         return self.reverse_mapping()
 
+    def get_all_provides_transitively(self, resource: ResourceIdStr | set[ResourceIdStr]) -> list[ResourceIdStr]:
+        """
+        This method returns all the provides (transitively) of the given resource.
+
+        :param resource: The resource or set of resources for which the provides have to be resolved transitively.
+        """
+        input_set = {resource} if isinstance(resource, str) else set(resource)
+        work: SimpleQueue[ResourceIdStr] = SimpleQueue()
+        for elem in input_set:
+            work.put_nowait(elem)
+        provides_mapping = self.provides_view()
+        # Use a dict here to not lost the order of the elements.
+        result: dict[ResourceIdStr, None] = {}
+        while not work.empty():
+            current_resource = work.get()
+            result[current_resource] = None
+            provides = provides_mapping.get(current_resource, set())
+            for elem in provides:
+                work.put_nowait(elem)
+        for elem in input_set:
+            result.pop(elem, None)
+        return list(result.keys())
+
 
 @dataclass(frozen=True)
 class ResourceDetails:
     resource_id: ResourceIdStr
     attribute_hash: str
     attributes: Mapping[str, object] = dataclasses.field(hash=False)
+    status: const.ResourceState
```

Comment:
So, as mentioned in the other thread: I do not think this belongs here. This is the old-style resource state we're trying to replace. That said, we can consider to add it now and to drop it when we make the scheduler the authority for resource state (#8187?). If we decide to take that approach, we should add a FIXME here.

Overall, I think I would prefer to drop it now already, and simply populate the `ResourceStatus` when we read from the database.

@wouterdb WDYT?

Reply:
Concretely, I think that the difference between old resource states `undefined` and `skipped_for_undefined` are the following in the new resources states:
- `undefined`: `status=UNDEFINED, blocked=YES`
- `skipped_for_undefined`: `status=UP_TO_DATE/HAS_UPDATE, blocked=YES`

Reply:
We handle it now or later, no strong preference.

Reply:
Added to #8187